Histo Branch 2

pchelle opened the pull request:
Commits include Abdullah histoBranch pull request

pchelle committed:
Add utils function ifequal

pchelle committed:
Simplify themes file saving information of default themes in data

pchelle committed:
Simplify the plot config when no mapping is provided

pchelle committed:
Add data to be accessed and used in vignettes in data folder

abdullahhamadeh committed:
Create superclass for datamapping-xy that only takes x

abdullahhamadeh committed:
Make XYDataMapping a child of XDataMapping

abdullahhamadeh committed:
Fix spacing

abdullahhamadeh committed:
Create datamapping for histograms, child of xdatamapping

abdullahhamadeh committed:
Delete datamapping-x.R

abdullahhamadeh committed:
make datamapping-xy a parent class

abdullahhamadeh committed:
Update datamapping-timeprofile.R

abdullahhamadeh committed:
Update datamapping-groupings.R

abdullahhamadeh committed:
Update datamapping-xygroup.R

abdullahhamadeh committed:
Update plotconfiguration.R

abdullahhamadeh committed:
Update plotconfiguration-legend.R

abdullahhamadeh committed:
Update plotconfiguration-label.R

abdullahhamadeh committed:
Update datamapping-xygroup.R

abdullahhamadeh committed:
Update datamapping-histogram.R

abdullahhamadeh committed:
updated histogram plot

abdullahhamadeh committed:
Update plotconfiguration-histogram.R

abdullahhamadeh committed:
Update plotconfiguration-histogram.R

abdullahhamadeh committed:
Update plot-histogram.R

abdullahhamadeh committed:
Update plotconfiguration-histogram.R

abdullahhamadeh committed:
Update plot-histogram.R

abdullahhamadeh committed:
Update plotconfiguration-histogram.R

pchelle committed:
Simplify watermark workflow

abdullahhamadeh committed:
update description

abdullahhamadeh committed:
added aggregation input class

user now has option of entering an aggregation input object containing function, function name, aggregation units and dimensions.  if not supplied, user may also input these individually in vector form, as before.

abdullahhamadeh committed:
updated description

pchelle committed:
NEWS.md is the accepted name for release notes (remove check warning)

pchelle committed:
Fix warnings in devtools::check()

pchelle committed:
Fix input position issue leading to wrong mapping and vignette crash

pchelle committed:
Resolve merge conflicts

pchelle committed:
Implement dummy aes variable on time profile configuration

pchelle committed:
Fix vignettes to work on devtools::check()

abdullahhamadeh committed:
Update time-profile-vignette.Rmd

msevestre reviewed (commented):
@pchelle Can you run the tidy style on the package? The indentation and spacing is weird at times.

Also there are methods with lots of empty spaces. Can we clean this?

msevestre commented on R/plot-histogram.R:
```diff
@@ -0,0 +1,57 @@
+  validateIsOfType(dataMapping, HistogramDataMapping)
+  validateIsOfType(plotConfiguration, HistogramPlotConfiguration)
+
+
```
remove spacing

pchelle committed:
Fixes #88 swap grouping and groupmapping

pchelle committed:
Fixes #88 implement swap into XYG mapping

pchelle committed:
Simplify for loop in aggregation summary

pchelle committed:
Remove spacing from plot-histogram

pchelle commented on R/plotconfiguration-histogram.R:
```diff
@@ -0,0 +1,142 @@
+        show.legend = TRUE,
+        binwidth = binWidth,
+        bins = bins,
+        alpha=.5
```
alpha is a parameter defining the transparency of shaded areas
The name is commonly used by both Matlab and R
I think the goal is to prevent one histogram to hide another
I will be adding default sizes and alphas in Theme

pchelle committed:
Add size, alpha and histogram properties to themes

pchelle committed:
Document utils

pchelle committed:
Documentation update on addition of theme properties

pchelle committed:
Fixes #88 Update of documentation on swapping

pchelle committed:
Styler update

pchelle committed:
Error-check method for dataMapping: are map variables included in data

pchelle committed:
Convert mapping to list usable by aes_string

pchelle committed:
Add check map data method + include dummy variable for aesthetics

pchelle committed:
Document utils

pchelle committed:
Integrate swapping to PKRatio

pchelle committed:
Add size to grouping options

pchelle committed:
Method initializing an alternative format for metaData

pchelle committed:
Integrate new mapping methods and swapping to time profile

pchelle committed:
Add LLOQ to time profile plot

pchelle committed:
Remove debugging code

pchelle committed:
Add test examples to dev for time profiles

pchelle committed:
Format fixes

pchelle committed:
Styler update

pchelle committed:
Update vignette code

pchelle committed:
Documentation title for %||%

pchelle commented:
In the last commits:
there are updates based on your comments + 
integration of the swap between Grouping and GroupMapping as suggested by Jurgen + 
removal of getMapData method, replaced it by checkMapData which has 2 steps: 
1) check if the variables are included in the data 
2) output data.frame with consistent variable names between x, y and groups that can be used by aes_string (which is cleaner than aes ggplot method)

However, I haven't checked and updated most of the code for histograms yet
Let me know if you have any question on those new commits

abdullahhamadeh committed:
Revert "Update time-profile-vignette.Rmd"

This reverts commit 5e22eacfef7bb69c0392842dbd11c36add233384.

abdullahhamadeh committed:
Merge branch 'histo-branch2' of https://github.com/pchelle/TLF-Library into pr/85

abdullahhamadeh committed:
Merge branch 'develop' into pr/85

pchelle committed:
Remove getMapData method
